cmd/manager/main.go

```diff
@@ -83,6 +98,20 @@ func main() {
 		exit(err)
 	}
 
+	ctx := context.TODO()
+
+	// Create Service object to expose the metrics port.
+	service, err := metrics.ExposeMetricsPort(ctx, metricsPort)
+	if err != nil {
+		log.Errorf("Unable to expose metrics port: %v", err)
+	}
+
+	// Create a serviceMonitor for the namespace, and register the service created above to it.
+	_, err = metrics.CreateServiceMonitors(cfg, namespace, []*v1.Service{service})
```

Comment:
Maybe this is my misunderstanding, but shouldn't we be using something as a result of this to instrument the code somehow? It's not clear to me what metrics are actually being collected.

Reply:
Prometheus library by default exposes some GoLang metrics. So right now, with this code the generic GoLang metrics would get exposed. But as you said, in case of custom metrics, we have to do the additional process of registering them with prometheus and then creating these resources (service and service monitor) so that they get published. I have not included that in this PR.

Reply:
That makes sense, essentially we are just exposing an endpoint that returns generic metric data about the running binary then? So what metrics does this actually provide? Do we have docs or examples from Prometheus or something similar? Can we add docs around what we are exposing and how it can be accessed (an example, at least)?

Reply:
Yes, we are only exposing the endpoint as of now. The GoLang metrics actually do not provide anything useful to us. These are Go specific system related metrics like `number of heap bytes obtained from system`, `number of seconds since garbage collection` and so on. Maybe these can be considered redundant data of no use.

Reply:
That's not necessarily not useful, it just depends on what our use case is.

Reply:
Yeah, Ill make a documentation on what metrics are being collected by prometheus so that it may be useful.